src/wp-includes/theme.php

```diff
@@ -840,6 +841,26 @@ function switch_theme( $stylesheet ) {
 	$new_theme->delete_pattern_cache();
 	$old_theme->delete_pattern_cache();
 
+	// Set autoload=no for the previous all themes.
+	$theme_mods_compare_value = '%' . $wpdb->esc_like( 'theme_mods_' ) . '%';
+	$theme_mods_options       = $wpdb->get_results( $wpdb->prepare( "SELECT option_name FROM $wpdb->options WHERE autoload = 'yes' AND option_name LIKE %s", $theme_mods_compare_value ) );
```

Comment:
Nitpick: you can do the `esc_like()` directly inside the `prepare()` method and avoid creating an extra variable that is only used once, ex:

```suggestion
	$theme_mods_options = $wpdb->get_results( $wpdb->prepare(
		"SELECT option_name FROM $wpdb->options WHERE autoload = 'yes' AND option_name LIKE %s",
		'%' . $wpdb->esc_like( 'theme_mods_' ) . '%'
	) );
```

Reply:
Also, this could use `$wpdb->get_col()` instead of `$wpdb->get_results()`